File: front/src/types/api/usage.ts
export type Usage = {
  use_type: string;
  use_date: string;
  note?: string;
  cook_name?: string;
  cook_category?: string;
  limit?: string;
  eat_rate?: number;
  use_stocks: Array<{ id: number, use_rate: number }>;
};

File: api/app/models/usage.rb
class Usage < ApplicationRecord
  belongs_to :stock, class_name: 'Stock', :foreign_key => 'stock_id'
  belongs_to :cook, class_name: 'Stock', :foreign_key => 'cook_id', optional: true
  validates :stock_id, presence: true
  validates :use_date, presence: true
  validates :use_type, presence: true
  validates :use_rate, presence: true,
                       numericality: {
                         greater_than_or_equal_to: 1,
                         less_than_or_equal_to: 100
                       }
  validates :note, length: { maximum: 50 }
end
